spec/unit/plugins/linux/network_spec.rb

```diff
@@ -1335,6 +1336,74 @@
         end
       end
 
+      # notes:
+      # route 172.16.19.39 is the addr of tun0 ; see :linux_ifconfig at top of file
+      # route 10.116.201.0/24 is discarded because it is not a default route
+      # route 192.168.0.0/24 is discarded because src 192.168.0.2 does not match eth0 ("Skipping route entry whose src does not match the interface IP")
+      # route 172.16.19.39 is chosen over 10.116.201.254 because its metric is lower (10 < 20)
+      describe "when there's a source field in a local route entry but no dev in the route" do
+        let(:linux_ip_route) do
+          <<~EOM
+            default proto bird src 172.16.19.39 metric 10 \\      nexthop via 10.1.81.1 dev eth0 weight 20
+            default via 10.116.201.254 dev eth0 metric 20 src 10.116.201.74
+            10.116.201.0/24 dev eth0  proto kernel  src 10.116.201.76
+            192.168.0.0/24 dev eth0  proto kernel  src 192.168.0.2
+          EOM
+        end
+        let(:linux_ip_route_inet6) { "" }
+
+        it "maps the no-dev route and sets ipaddress" do
+          plugin.run
+          expect(plugin["ipaddress"]).to eq("172.16.19.39")
+        end
+      end
+
+      # notes:
+      # route 172.16.19.39 is the addr of tun0 ; see :linux_ifconfig at top of file
+      # route 10.116.201.1 discarded because src 192.116.201.1 does not match eth0 ("Skipping route entry whose src does not match the interface IP")
```

Comment:
That doesn't match the route:
```
default via 10.116.201.1 dev eth0 metric 10 src 10.116.201.1
```
Did you mean "src 10.116.201.1 does not match..." ?

Reply:
oops , that line shouldn't be there sorry. its actually not discarded, just not chosen because it is higher priority. Updated the comments to reflect this